### PR TITLE
Add throwOnFail to DoesCIExist

### PR DESCRIPTION
**Pull request by istairbn**
DoesCIExist returns a boolean for existence. This is suitable for a conditional check, but if you want to Stop on a False requires an additional task. 

Added the throwOnFail - if a False is returned then this task will Fail. This allows the Template creator to use this task as a hard stop, rather than as a conditional or needing a second task to confirm success later.

### CODE REVIEW

File: src/main/resources/synthetic.xml [hunk 1]
```diff
@@ -11,7 +11,6 @@
     <type type="xldeploy.Task" extends="xlrelease.PythonScript" virtual="true">
         <property name="iconLocation" default="xldeploy/xldeploy-icon-black-small.png" hidden="true" />
         <property name="taskColor" hidden="true" default="#68B749" />
-
```
Comment (jdewinne) [inline src/main/resources/synthetic.xml:14]: Is there a reason why all these newlines are removed? Shouldn't they stay in the `xml` for readability?

Comment (istairbn) [reply src/main/resources/synthetic.xml:14]: Valid point - however the XML wasn't consistently spaced for readability: some types were and some were not. I simply set all types to non-spaced blocks to at least keep it consistent.

Comment (jdewinne) [reply src/main/resources/synthetic.xml:14]: Is there a way you can put the newlines back? I'm especially concerned about the distinction between input properties and output properties.

Comment (istairbn) [reply src/main/resources/synthetic.xml:14]: Sure

Comment (jdewinne) [reply src/main/resources/synthetic.xml:14]: Great. Let me know if you're ready, and I'll merge.

**Pull request closed without merging**